compiler/testData/diagnostics/tests/classObjects/typeParametersInAnnonymousObject.fir.kt

```diff
@@ -33,7 +33,7 @@ inline fun <reified T> case_5() {
 
     val z = x.test()
 
-    if (z is T) {
+    if (<!USELESS_IS_CHECK!>z is T<!>) {
```

Comment:
This and below (`...after`), return type of `x.test()` is `T`, hence `z is T`. FE 1.0's subtype check regards `T <\: T`, while NI type check (w/ `ConeInferenceContext`) regards `T <: T`. I think NI type check seems right.

Reply:
Yes, the result is different, i.e., `T <\: SomeOtherTypeParameter`. I guess using the reified `T` is the intention of line 30, so added another set of object with type parameter below (and related files).